### PR TITLE
Add tf-session-runs-view; fix py_tests

### PR DESCRIPTION
**Pull request by caisq**
Also in this change:
* Move the "Waiting for first Session.run() to connnect..." message to a
  separate dialog.
* Make all py_tests pass. They have been failing previously due to
  changes to the DebuggerDataServer and DebuggerDataStreamHandler
  classes that broke assumption from the nanchucks mode.

### CODE REVIEW

**Comment by caisq** [comment]
Initial waiting dialog:
tdp_waiting

**Comment by caisq** [comment]
Session runs view (see bottom-left corner):
tdp_session_runs

File: tensorboard/plugins/debugger/debugger_server_lib.py [hunk 1]
```diff
@@ -214,6 +212,10 @@ def on_core_metadata_event(self, event):
     # Wait for acknowledgement from client.
     tf.logging.warning("Waiting for client ack (meta)...")
     # Blocks until an item is got.
+    # TODO(cais): Before merging to master, the following get_incoming() call
+    # must be made conditional on whether tensorboard is operating under
+    # the nancucks mode or the general TPD mode, because the nanchucks mode
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/debugger_server_lib.py]: Maybe point to docs that clarify the difference?

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-session-runs-view.html [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      const targetsCell = document.createElement('td');
+      targetsCell.textContent = JSON.stringify(sessionRun.targets);
+      if (isActive) {
+        targetsCell.style.fontWeight = 'bold';
+      }
+      const countCell = document.createElement('td');
+      countCell.textContent = count
+
+      if (isActive) {
+        countCell.style.fontWeight = 'bold';
```
Comment (chihuahua) [inline tensorboard/plugins/debugger/tf_debugger_dashboard/tf-session-runs-view.html]: You could instead define a CSS class for active count cells. For instance,

CSS: 
```css
:host .active-cell {
  font-weight: bold;
}
```

In JS,
```js
Polymer.dom(countCell).classList.add('active-cell');
```

File: third_party/workspace.bzl [hunk 3]
```diff
@@ -31,11 +31,11 @@ def tensorboard_workspace():
   native.http_archive(
       name = "protobuf",
       urls = [
-          "http://mirror.bazel.build/github.com/google/protobuf/archive/2b7430d96aeff2bb624c8d52182ff5e4b9f7f18a.tar.gz",
-          "https://github.com/google/protobuf/archive/2b7430d96aeff2bb624c8d52182ff5e4b9f7f18a.tar.gz",
+          "http://mirror.bazel.build/github.com/google/protobuf/archive/v3.4.1.tar.gz",
```
Comment (chihuahua) [inline third_party/workspace.bzl]: SG, lets include this fix for now.

File: tensorboard/plugins/debugger/debugger_server_lib.py (hunk 1 above)
Comment (caisq) [reply tensorboard/plugins/debugger/debugger_server_lib.py]: This is not well documented currently. We should discuss solutions to this TDP-nanchucks compatibility problem.

File: third_party/workspace.bzl (hunk 3 above)
Comment (caisq) [reply third_party/workspace.bzl]: Ack.

File: tensorboard/plugins/debugger/tf_debugger_dashboard/tf-session-runs-view.html (hunk 2 above)
Comment (caisq) [reply tensorboard/plugins/debugger/tf_debugger_dashboard/tf-session-runs-view.html]: Done. Thanks.

**Pull request merged**